src/transformers/models/bert/modeling_tf_bert.py

```diff
@@ -821,7 +821,7 @@ def call(
             )
         else:
             extended_attention_mask = tf.reshape(
-                inputs["attention_mask"], (attention_mask_shape[0], 1, 1, attention_mask_shape[1])
+                inputs["attention_mask"], (attention_mask_shape[0], 1, -1, attention_mask_shape[1])
```

Comment:
I don't think this is the right fix. When the attention mask is going to be 3d, we should use `attention_mask_shape[2]` for the last dimension. The generic fix should thus be:
```suggestion
                inputs["attention_mask"], (attention_mask_shape[0], 1, -1, attention_mask_shape[-1])
```

Reply:
thanks for your comment, and you are right, this is a lazy way, we can make it the same way as pytorch version:
line 262 in modeling_utils
        if attention_mask.dim() == 3:
            extended_attention_mask = attention_mask[:, None, :, :]
        elif attention_mask.dim() == 2:
            .......
how do you think?

Reply:
and yes, it should be a common function for all transformer models in tf version